packages/core/useStepper/index.ts

```diff
@@ -0,0 +1,83 @@
+import { computed, ref } from 'vue-demi'
+
+export function useStepper<T extends string>(steps: readonly T[], initial?: T) {
+  const index = ref(steps.indexOf(initial ?? steps[0]))
+  const current = computed(() => steps[index.value])
+  const isFirst = computed(() => index.value === 0)
+  const isLast = computed(() => index.value === steps.length - 1)
+
+  function backTo(step: T) {
+    if (currentStepIsAfter(step))
+      goTo(step)
+  }
+
+  function goTo(step: T) {
+    index.value = steps.indexOf(step)
+  }
+
+  function next() {
+    if (isLast.value)
+      return
+
+    index.value++
+  }
+
+  function previous() {
+    if (isFirst.value)
+      return
+
+    index.value--
+  }
+
+  function isNext(step: T) {
+    return steps.indexOf(step) === index.value + 1
+  }
+
+  function isPrevious(step: T) {
+    return steps.indexOf(step) === index.value - 1
+  }
+
+  function currentStepIs(step: T) {
+    return steps.indexOf(step) === index.value
+  }
+
+  function currentStepIsAfter(step: T) {
+    return index.value > steps.indexOf(step)
+  }
+
+  function currentStepIsBefore(step: T) {
+    return index.value < steps.indexOf(step)
+  }
+
+  function completed(step: T) {
+    return currentStepIsAfter(step)
+  }
+
+  function active(step: T) {
+    return currentStepIs(step)
+  }
+
+  function todo(step: T) {
+    return currentStepIsBefore(step)
+  }
```

Comment:
I not sure if I understand what's these three are for.

Reply:
These are aliases, which are useful in some contexts when using the literal function name (eg. `currentStepIsBefore`) makes things more confusing than they should (eg. `todo`).

I would understand if you'd rather not have them. They helped make the code clean when I needed to determine if a step was yet to be done, I could use `todo` instead of having to wrap my head around what was `currentStepIsBefore` doing. What do you think?

Reply:
Good point. I removed them. 👍